crypto/crypto.go

```diff
@@ -17,6 +18,7 @@ const (
 
 // TODO - encryption mode consts
 
+// Encrypted represents a ciphertext with related inputs
 type Encrypted struct {
 	Ciphertext string
 	Mode       string
```

Comment:
This is my fault, but we should probably be explicit with the Encrypted.Mode field and make its type Mode instead of string.